optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -94,11 +93,11 @@ public String getPhaseTypeString() {
     @Override
     public void solve(DefaultSolverScope<Solution_> solverScope) {
         PartitionedSearchPhaseScope<Solution_> phaseScope = new PartitionedSearchPhaseScope<>(solverScope);
-        phaseStarted(phaseScope);
         List<Solution_> partList = solutionPartitioner.splitWorkingSolution(
                 solverScope.getScoreDirector(), runnablePartThreadLimit);
         int partCount = partList.size();
         phaseScope.setPartCount(partCount);
+        phaseStarted(phaseScope);
```

Comment:
Thanks to this change, listeners will be able to read the `partCount` when the phase starts.